src/ILLink.Tasks/LinkTask.cs

```diff
@@ -303,6 +309,9 @@ protected override string GenerateResponseFileCommands ()
 			if (OutputDirectory != null)
 				args.Append ("-out ").AppendLine (Quote (OutputDirectory.ItemSpec));
 
+			if (NoWarn != null)
+				args.Append ($"--nowarn \"{NoWarn}\"");
```

Comment:
```suggestion
				args.Append ("--nowarn ").AppendLine (Quote (NoWarn));
```
Mostly to follow conventions. The important bit is that it needs whitespace after the option, to prevent being combined with the next option.

Could you also add a test in `ILLink.Tasks.Tests.cs`?